addons/purchase_stock/models/stock_rule.py

```diff
@@ -51,6 +51,7 @@ def _run_buy(self, procurements):
             partner = supplier.name
             # we put `supplier_info` in values for extensibility purposes
             procurement.values['supplier'] = supplier
+            procurement.values['propagate'] = rule.propagate
```

Comment:
What is the purpose to put 'propagate' in values. You probably won't use it outside the scope of the _run_method. And at each step it will be erase by the new rule propagate.

Reply:
We have to add new value in obj ( propagate ) is it okey .. .because we have multiple values to be passed in rules
group by